change/@ni-nimble-components-04e43df1-a30f-47c3-a47f-2ecd40655db7.json

```diff
@@ -0,0 +1,7 @@
+{
```

Comment:
> @rajsite @atmgrifter00 - Could you take a look at this draft PR? If you are happy with the direction, I'll write some unit tests for the selection manager classes I've created.

I don't have any immediate major concerns I can think of with this direction from an initial scan. It looks like it helps encapsulate some of the selection behavior and state nicely. 👍

Reply:
**TLDR** - I decided not to write unit tests for the selection manager classes I created here because they seemed expensive to write and were only going to duplicate test coverage of things tested from the table's selection tests.

**Longer version**
I started looking into writing unit tests for the new selection manager classes. I naively thought it would be easy to write a bunch of unit tests around a well-scoped API, but as I started writing those tests, I quickly realized that was not the case. Because the selection manager base class takes in an instance of `TanStackTable`, I would need to create a TanStack table with the same options as we expect the nimble-table to configure on TanStack. As I started pulling that thread, I realized that the value in unit tests wasn't great because part of the behavior that should be tested is that TanStack has the correct configuration given the nimble-table's configuration and that the nimble-table reacts to certain callbacks on the TanStack table correctly. The selection manager code is tested by `table-selection.spec.ts`, so trying to mock all the same scenarios through direct manipulation of TanStackTable within additional unit tests seems like low ROI.

Reply:
Makes sense to me! Thanks for the explanation